polkadot/runtime/parachains/src/hrmp.rs

```diff
@@ -1215,10 +1219,19 @@ impl<T: Config> Pallet<T> {
 
 	/// Returns HRMP watermarks of previously sent messages to a given para.
 	pub(crate) fn valid_watermarks(recipient: ParaId) -> Vec<BlockNumberFor<T>> {
-		HrmpChannelDigests::<T>::get(&recipient)
+		let mut valid_watermarks: Vec<_> = HrmpChannelDigests::<T>::get(&recipient)
 			.into_iter()
 			.map(|(block_no, _)| block_no)
-			.collect()
+			.collect();
+
+		// The current watermark will remain valid until updated.
+		if let Some(last_watermark) = HrmpWatermarks::<T>::get(&recipient) {
```

Comment:
same as in the inclusion_emulator, do we really need this?
This is only really used in the inclusion emulator

Reply:
Okay, the "problem" is this pruning: https://github.com/paritytech/polkadot-sdk/blob/d3128dc679a089750a8569acc6e0834787a7b87c/polkadot/runtime/parachains/src/hrmp.rs#L1312

But yeah, your solution is correct.